### PR TITLE
Add vaapi 4:2:0 12bit decode support

### PR DESCRIPTION
**Pull request by sunxinpeng**
Signed-off-by: Xinpeng Sun <xinpeng.sun@intel.com>

### CODE REVIEW

**Comment by zhoujd** [comment]
This PR failed on VP9 10bit decode in https://media-ci.ostc.intel.com/build/389884

File: libavcodec/hevcdec.c [hunk 1]
```diff
@@ -438,6 +438,9 @@ static enum AVPixelFormat get_format(HEVCContext *s, const HEVCSPS *sps)
     case AV_PIX_FMT_YUV444P12:
 #if CONFIG_HEVC_NVDEC_HWACCEL
         *fmt++ = AV_PIX_FMT_CUDA;
+#endif
+#if CONFIG_HEVC_VAAPI_HWACCEL
```
Comment (fulinjie) [inline libavcodec/hevcdec.c:442]: It seems to be the decode support for 4:2:0 12bit, why the AV_PIX_FMT_VAAPI is enabled for YUV444P12?

File: libavcodec/vaapi_hevc.c [hunk 2]
```diff
@@ -552,6 +552,9 @@ VAProfile ff_vaapi_parse_hevc_rext_profile(AVCodecContext *avctx)
     else if (!strcmp(profile->name, "Main 4:4:4 10") ||
              !strcmp(profile->name, "Main 4:4:4 10 Intra"))
         return VAProfileHEVCMain444_10;
+    else if (!strcmp(profile->name, "Main 12") ||
+	     !strcmp(profile->name,"Main 12 Intra"))
```
Comment (fulinjie) [inline libavcodec/vaapi_hevc.c:556]: indentation issues.

**Comment by fulinjie** [comment]
Seems to break vp9 decoding for vaapi iHD

**Comment by zhoujd** [comment]
I am closing this PR, please submit new PR if you have the update of the patch

**Pull request closed without merging**